File: server/src/SyncVR.Server/Models/FibonacciQuery.cs
namespace SyncVR.Server.Models;

public record FibonacciQuery
{
    /// <summary>
    /// The unique query identifier.
    /// </summary>
    public Guid Id { get; init; }

    /// <summary>
    /// The timestamp when the query was made.
    /// </summary>
    public DateTime CreatedAt { get; init; }

    /// <summary>
    /// The Fibonacci number at the requested position.
    /// </summary>
    public ulong Value { get; init; }

    /// <summary>
    /// The requested position.
    /// </summary>
    public int Position { get; init; }

    /// <summary>
    /// An anonymous user identifier.
    /// </summary>
    public string ClientId { get; init; } = null!;
}

File: client/src/models/fibonacci-query.ts
export type FibonacciQueryDbo = {
  id: string
  createdAt: string
  clientId: string
  position: number
  value: number
}

export class FibonacciQuery {
  constructor(
    public id: string,
    public createdAt: Date,
    public clientId: string,
    public position: number,
    public value: number
  ) {}

  static fromJson(data: FibonacciQueryDbo) {
    return new FibonacciQuery(
      data.id,
      new Date(data.createdAt),
      data.clientId,
      data.position,
      data.value
    )
  }
}
